gutenberg.php

```diff
@@ -16,20 +16,144 @@
 defined( 'GUTENBERG_DEVELOPMENT_MODE' ) or define( 'GUTENBERG_DEVELOPMENT_MODE', true );
 ### END AUTO-GENERATED DEFINES
 
-gutenberg_pre_init();
+// Minimum supported version. Has to match "Requires at least" header above.
+if ( ! defined( 'GUTENBERG_MIN_WP_VERSION' ) ) {
+	define( 'GUTENBERG_MIN_WP_VERSION', '5.7' );
+}
+
+// Next unsupporetd version.
+if ( ! defined( 'GUTENBERG_NEXT_UNSUPPORTED_WP_VERSION' ) ) {
+	// The next major WordPress version that is not supported yet.
+	// Generally that would be the current WP trunk + 1. This will allow running of
+	// the currently released Gutenberg while the next WP version is under development,
+	// but not after it has been released.
+	// For example if the next unsupported version is 6.0, Gutenberg will still run
+	// in 6.0-alpha-12345, 6.0-beta1, etc. but not in 6.0, 6.0.1, 6.0.1-alpha, etc.
+	// Also note that `version_compare()` considers 1 < 1.0 < 1.0.0 (before suffixes)
+	// so 6.0 < 6.0.0-alpha and 6.0-beta < 6.0.0-alpha are both true. For that reason
+	// the constant should be set to major (two digits) WordPress version.
+	define( 'GUTENBERG_NEXT_UNSUPPORTED_WP_VERSION', '6.0' );
```

Comment:
It might be risky, but maybe we could also auto-generate this constant from `Tested up to`:
https://github.com/WordPress/gutenberg/blob/1d0f3c415347cd6550e84e02ff77211e6da06277/readme.txt#L3

Reply:
Yea, was thinking the same. Generally the `Tested up to` is set to the current WP version, so will need to increment it.

Was wondering, what would be a good rule-of-thumb max WP version? As far as I see fixes and new features that get added to Gutenberg while WP is in alpha are generally synced (merged) with core trunk before WP is released, i.e. in the same WP version. The question is: should it stop loading Gutenberg if WP is newer than the `Tested up to`, or should it just show a warning to update Gutenberg in  `Tested up to` +1, and stop loading in `Tested up to` +2?

Reply:
> It might be risky, but maybe...

Looking a bit more, actually not sure this would work well. The problem is that if Gutenberg is released during WP alpha or (early?) beta, this should be set to `Tested up to` +1. However if Gutenberg is released during WP RC, this should probably be set to `Tested up to` +2 as any new features or bugfixes will most likely not be added to the WP RC branch (branching of WP trunk usually happens shortly after RC1). So seems this would have to be set manually. Of course there will be a lengthily explanation in inline comments about how to do it.

Reply:
Right, it's very difficult to reason how that could work. I need to think a bit more about all the relations between the Gutenberg versions backported to the given major version of WordPress core, `Tested up to`, the release process, etc. The goal from my perspective would be to automate as much as possible so people don't have to go through the same painful process again reasoning about all the implications 😄

Reply:
> people don't have to go through the same painful process again reasoning about all the implications

Yea, automating the decision would be best. Not sure if that would be possible though, seems too many variables and possible exceptions. The next best thing would be to add as much help and explanations as possible. Frankly I don't mind having "couple pages long" docblock there.